engine/src/main/java/org/apache/apex/engine/plugin/ApexPluginManager.java

```diff
@@ -51,7 +52,7 @@
  * A default implementation for ApexPluginDispatcher. It handler common tasks such as per handler
  * registration. actual dispatching is left for classes extending from it.
  */
-public abstract class ApexPluginManager extends AbstractService
+public abstract class ApexPluginManager extends AbstractService implements ApexPluginDispatcher
```

Comment:
For consistency with how other plugin classes are named how about calling this AbstractApexPluginDispatcher